src/kitovu/sync/syncing.py

```diff
@@ -65,3 +73,19 @@ def start(connection_settings: ConnectionSettings) -> None:
             print(f'Local digest: {digest}')
 
     plugin.disconnect()
+
+
+def config_error(config_file: typing.Optional[pathlib.Path]) -> typing.Union[str, None]:
+    """Validates the given configuration file.
+
+    Returns either an error message or None if it's valid."""
```

Comment:
Makes me wonder whether it should raise an exception object (like `InvalidSettingsError`) instead. That has several benefits:

- You can reraise it as `utils.UsageError` in `cli.py` causing click to automatically do the right thing (like writing it to stderr and not stdout :wink:)
- It can have some kind of `.errors` attribute which is much easier to test, so you don't have to parse the formatted error string again for testing.

What do you think?